openedx/core/djangoapps/discussions/tests/test_views.py

```diff
@@ -93,3 +127,298 @@ class CourseStaffAuthorizedTest(AuthorizedApiTest):
+        response = self._post(payload)
+        data = response.json()
+        assert response.status_code == self.expected_response_code
+        return data
+
+    def test_get_nonexistent_with_defaults(self):
+        """
+        If no record exists, defaults should be returned.
+        """
+        response = self._get()
+        self._assert_defaults(response)
+
+    def test_post_everything(self):
+        """
+        API should accept requests to update _all_ fields at once
+        """
+        data = self._setup_lti()
+        assert data['enabled']
+        assert data['provider_type'] == 'piazza'
+        assert data['providers']['available']['piazza'] == PROVIDER_FEATURE_MAP['piazza']
+        assert data['lti_configuration'] == DATA_LTI_CONFIGURATION
+        assert len(data['plugin_configuration']) == 0
+        assert len(data['lti_configuration']) > 0
+        response = self._get()
+        # the GET should pull back the same data as the POST
+        response_data = response.json()
+        assert response_data == data
+
+    def test_post_invalid_key(self):
+        """
+        Unsupported keys should be gracefully ignored
+        """
+        payload = {
+            'non-existent-key': 'value',
+        }
+        response = self._post(payload)
+        self._assert_defaults(response)
+
+    def test_configuration_valid(self):
+        """
+        Check we can set basic configuration
+        """
+        provider_type = 'piazza'
+        payload = {
+            'enabled': True,
+            'provider_type': provider_type,
+            'plugin_configuration': {
+                'key': 'value',
+            },
+        }
+        self._post(payload)
+        response = self._get()
+        data = response.json()
+        assert data['enabled']
+        assert data['provider_type'] == provider_type
+        assert data['plugin_configuration'] == payload['plugin_configuration']
+
+    @ddt.data(
+        {
+            'enabled': 3,
+        },
+    )
+    def test_configuration_invalid(self, payload):
+        """
+        Check validation of basic configuration
+        """
+        with self.assertRaises(ValidationError):
+            response = self._post(payload)
+        response = self._get()
+        self._assert_defaults(response)
+
+    def test_post_lti_valid(self):
+        """
+        Check we can set LTI configuration
+        """
+        provider_type = 'piazza'
+        for key, value in DATA_LTI_CONFIGURATION.items():
+            payload = {
+                'enabled': True,
+                'provider_type': provider_type,
+                'lti_configuration': {
+                    key: value,
+                }
+            }
+            response = self._post(payload)
+            response = self._get()
+            data = response.json()
+            assert data['enabled']
+            assert data['provider_type'] == provider_type
+            assert data['lti_configuration'][key] == value
```

Comment:
Question: will `data['lti_configuration']` contain other keys as well? e.g. The loop will run for each item in `DATA_LTI_CONFIGURATION` and will post/get for each key/value.